src/cmd/ksh93/sh/path.c

```diff
@@ -1094,15 +1094,20 @@ pid_t path_spawn(Shell_t *shp, const char *opath, char **argv, char **envp, Path
     }
     if (pid > 0) return pid;
 retry:
-    switch (shp->path_err = errno) {
+    shp->path_err = errno;
+    switch (shp->path_err) {
+        case EISDIR: {
+            return -1;
+        }
         case ENOEXEC:
         case EPERM: {
             // Some systems return EPERM if setuid bit is on.
             errno = ENOEXEC;
             if (spawn) {
-                if (shp->subshell) return (-1);
+                if (shp->subshell) return -1;
                 do {
-                    if ((pid = fork()) > 0) return (pid);
```

Comment:
It is a shame that these anti-patterns are so common in this project. I don't think the authors were deliberately trying to make the code hard to read but that was the net effect.